debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -189,54 +189,72 @@ protected void parseKeyFields(Struct key, Record r) {
      */
     protected void parseValueFields(Struct value, Record r) {
         Struct after = value.getStruct("after");
-        // TODO: error handle before is NULL
         Struct before = value.getStruct("before");
         
-        // in case of delete, the before struct contains the values required in cases like DELETE-INSERT conflicts(unique key columns)
-        if (r.op.equals("d") && before != null) {
-            after = before;
+        if (sourceType.equals("yb")) {
+            parseValueFieldsForYB(after, r);
+        } else {
+            parseValueFieldsForOthers(after, before, r);
         }
+    }
+
+    private void parseValueFieldsForYB(Struct after, Record r) {
         if (after == null) {
             return;
         }
 
         for (Field f : after.schema().fields()) {
-            Object fieldValue;
-            if (sourceType.equals("yb")){
-                // TODO: write a proper transformer for this logic
-                // values in the debezium connector are as follows:
-                // "val1" : {
-                //  "value" : "value for val1 column",
-                //  "set" : true
-                //}
-                Struct valueAndSet = after.getStruct(f.name());
-                if (r.op.equals("u")) {
-                    // in the default configuration of the stream, for an update, the fields in the after struct
-                    // are only the delta fields, therefore, it is possible for a field to not be there.
-                    if (valueAndSet == null){
-                        continue;
-                    }
-                } else if (r.op.equals("d") && valueAndSet == null){
-                    // in case of deletes we are using before struct which contains only delta for yb
+            // TODO: write a proper transformer for this logic
+            // values in the debezium connector are as follows:
+            // "val1" : {
+            //  "value" : "value for val1 column",
+            //  "set" : true
+            //}
+            Struct valueAndSet = after.getStruct(f.name());
+            if (r.op.equals("u")) {
+                // in the default configuration of the stream, for an update, the fields in the after struct
+                // are only the delta fields, therefore, it is possible for a field to not be there.
+                if (valueAndSet == null){
                     continue;
                 }
-                
-                if (!valueAndSet.getBoolean("set")){
+            } else if (r.op.equals("d") && valueAndSet == null){
```

Comment:
As of now, since we are not enabling before image, maybe we can just remove this block? 
Anyways, at the beginning of this func, if after == null, we return

Reply:
yeah, some leftover code. will remove it.